test/core/transport/chttp2/settings_timeout_test.cc

```diff
@@ -111,7 +111,7 @@ class Client {
     EventState state;
     grpc_tcp_client_connect(state.closure(), &endpoint_, pollset_set,
                             nullptr /* channel_args */, server_addresses->addrs,
-                            1000);
+                            2000);
```

Comment:
based on my testing (running bazel RBE from my workstation), increasing the timeout actually fixes the problem (I got 500 successful runs without a failure), but I'd like to understand more why increasing this timeout actually helps (are we just increasing the timeout blindly or is there some reasoning behind this?)

https://source.cloud.google.com/results/invocations/6a11ddab-7617-47d6-b1b5-74a63d3e17b2/targets/%2F%2Ftest%2Fcore%2Ftransport%2Fchttp2:settings_timeout_test/tests